tests/loom.rs

```diff
@@ -0,0 +1,38 @@
+#[cfg(loom)]
+#[cfg(test)]
+mod loom_tests {
+    // Evil hack to share CounterAddOp between
+    // unit tests and integration tests.
+    use left_right::Absorb;
+    include!("../src/utilities.rs");
+
+    use loom::sync::atomic::{AtomicI32, Ordering};
+    use loom::sync::Arc;
+    use loom::thread;
+
+    #[test]
+    fn read_before_publish() {
+        loom::model(|| {
+            let (mut w, r) = left_right::new::<i32, _>();
+            let answer = Arc::new(AtomicI32::new(0));
+            let answer_clone = answer.clone();
+
+            w.append(CounterAddOp(1));
+            w.publish();
+
+            let jh = thread::spawn(move || {
+                let val = *r.enter().unwrap();
+                answer_clone.store(val, Ordering::SeqCst);
+            });
+            w.publish();
+            w.append(CounterAddOp(1));
+
+            jh.join().unwrap();
+
+            // This is just to ensure the write handle is not dropped before the reader reads
```

Comment:
I'm not sure I understand this comment? `w` won't be dropped until the end of the scope anyway..

Reply:
That's a very good point 😅. I think I was afraid rust would think like, "Well this isn't used anymore, might as well clean up early" or something crazy. Updated in cc8711d6623a652b9b3327edf9c7a1269df77d29